crates/aptos/src/node/mod.rs

```diff
@@ -1170,30 +1170,48 @@ impl CliCommand<()> for RunLocalTestnet {
             }
 
             // Start the faucet
-            FaucetArgs {
-                address: "0.0.0.0".to_string(),
-                port: self.faucet_port,
-                server_url: rest_url,
-                mint_key_file_path: test_dir.join("mint.key"),
-                mint_key: None,
-                mint_account_address: None,
-                chain_id: ChainId::test(),
-                maximum_amount: None,
-                do_not_delegate: self.do_not_delegate,
-            }
-            .run()
-            .await;
-            Some(())
+            Some(
+                FaucetArgs {
+                    address: "0.0.0.0".to_string(),
+                    port: self.faucet_port,
+                    server_url: rest_url,
+                    mint_key_file_path: test_dir.join("mint.key"),
+                    mint_key: None,
+                    mint_account_address: None,
+                    chain_id: ChainId::test(),
+                    maximum_amount: None,
+                    do_not_delegate: self.do_not_delegate,
+                }
+                .run(),
+            )
         } else {
             None
         };
 
-        // Wait for an interrupt
-        let term = Arc::new(AtomicBool::new(false));
-        while !term.load(Ordering::Acquire) {
-            std::thread::park();
+        // Collect futures that should never end.
+        let mut futures: Vec<Pin<Box<dyn futures::Future<Output = ()> + Send>>> = Vec::new();
+
+        // This future just waits for the node thread.
+        let node_future = async move {
+            loop {
+                if node_thread_handle.is_finished() {
+                    return;
+                }
+                tokio::time::sleep(Duration::from_millis(500)).await;
+            }
```

Comment:
seems like you should just use join?

Reply:
Let me try that, but when I wrote this I think there was a good reason I didn't do this...

Reply:
Something about not blocking in futures.

Reply:
well this is a separate thread, so it should be perfectly safe... I wonder if there's an awaitable join?
https://docs.rs/async-thread/latest/async_thread/

that's not gonna work

do we have any utils we could add this too? no one should have have to see this code lol....

if you really wanna keep it around, let's just do the following:
```suggestion
            while !node_thread_handle.is_finished() {
                tokio::time::sleep(Duration::from_millis(10)).await;
            }
```

no need for the return statement and we can reduce the time between checks, since they are relatively cheap